### PR TITLE
Fix minor problems in NSX-V related resources

### LINKED ISSUE #376: Improvements needed for NSX-V DNAT
* in documentation example called "forIcmp", but is about "TCP"
* when external netw isn't right, error isn't precise:
```
vcd_nsxv_dnat.forIcmp: Creating...
Error: unable to make structure for API call: unable to identify vNic for network 'my-external-network' of type 'ext': [ENF] entity not found
  on main.tf line 212, in resource "vcd_nsxv_dnat" "forIcmp":
 212: resource "vcd_nsxv_dnat" "forIcmp" {
```
* Error got when for rule_tag entered string:
```
vcd_nsxv_dnat.forIcmp: Creating...
Error: error creating new NAT rule: error creating NAT rule:  The input request in invalid.
 - XStream unmarshalling exception; nested exception is com.thoughtworks.xstream.converters.ConversionException: 
---- Debugging information ----
cause-exception     : java.lang.NumberFormatException
cause-message       : For input string: "aaa1"
class               : java.lang.Long
required-type       : java.lang.Long
converter-type      : com.thoughtworks.xstream.converters.SingleValueConverterWrapper
wrapped-converter   : com.thoughtworks.xstream.converters.basic.LongConverter
line number         : 4
class[1]            : com.vmware.vshield.edge.nat.dto.NatRuleDto
converter-type[1]   : com.thoughtworks.xstream.converters.reflection.ReflectionConverter
class[2]            : com.vmware.vshield.edge.nat.dto.NatRules
version             : not available
-------------------------------
 - 
---- Debugging information ----
cause-exception     : java.lang.NumberFormatException
cause-message       : For input string: "aaa1"
class               : java.lang.Long
required-type       : java.lang.Long
converter-type      : com.thoughtworks.xstream.converters.SingleValueConverterWrapper
wrapped-converter   : com.thoughtworks.xstream.converters.basic.LongConverter
line number         : 4
class[1]            : com.vmware.vshield.edge.nat.dto.NatRuleDto
converter-type[1]   : com.thoughtworks.xstream.converters.reflection.ReflectionConverter
class[2]            : com.vmware.vshield.edge.nat.dto.NatRules
version             : not available
-------------------------------
 - For input string: "aaa1" (API error: 400)
  on main.tf line 212, in resource "vcd_nsxv_dnat" "forIcmp":
 212: resource "vcd_nsxv_dnat" "forIcmp" {
```
* !!! If rule_tag specified and any changes comes from UI, when always resource will be recreated as rule_tag don't match anymore:
```
  # vcd_nsxv_dnat.forIcmp must be replaced
-/+ resource "vcd_nsxv_dnat" "forIcmp" {
        description        = "a1"
        edge_gateway       = "vb_edge2"
      ~ enabled            = true -> false
      ~ id                 = "209088" -> (known after apply)
      ~ logging_enabled    = true -> false
        network_name       = "vb_externalNet2"
        network_type       = "ext"
        original_address   = "192.168.1.10"
        original_port      = "443"
        protocol           = "tcp"
      + rule_tag           = "65539" # forces replacement
      ~ rule_type          = "user" -> (known after apply)
        translated_address = "10.10.0.5"
        translated_port    = "8443"
    }
```

### LINKED ISSUE #377: Improvements needed for NSX-V SNAT
vsx Snat:
* Error got when for rule_tag entered string:
```
vcd_nsxv_dnat.forIcmp: Creating...
Error: error creating new NAT rule: error creating NAT rule:  The input request in invalid.
 - XStream unmarshalling exception; nested exception is com.thoughtworks.xstream.converters.ConversionException: 
---- Debugging information ----
cause-exception     : java.lang.NumberFormatException
cause-message       : For input string: "gaerre"
class               : java.lang.Long
required-type       : java.lang.Long
converter-type      : com.thoughtworks.xstream.converters.SingleValueConverterWrapper
wrapped-converter   : com.thoughtworks.xstream.converters.basic.LongConverter
line number         : 4
class[1]            : com.vmware.vshield.edge.nat.dto.NatRuleDto
converter-type[1]   : com.thoughtworks.xstream.converters.reflection.ReflectionConverter
class[2]            : com.vmware.vshield.edge.nat.dto.NatRules
version             : not available
-------------------------------
 - 
---- Debugging information ----
cause-exception     : java.lang.NumberFormatException
cause-message       : For input string: "gaerre"
class               : java.lang.Long
required-type       : java.lang.Long
converter-type      : com.thoughtworks.xstream.converters.SingleValueConverterWrapper
wrapped-converter   : com.thoughtworks.xstream.converters.basic.LongConverter
line number         : 4
class[1]            : com.vmware.vshield.edge.nat.dto.NatRuleDto
converter-type[1]   : com.thoughtworks.xstream.converters.reflection.ReflectionConverter
class[2]            : com.vmware.vshield.edge.nat.dto.NatRules
version             : not available
-------------------------------
 - For input string: "gaerre" (API error: 400)
  on main.tf line 212, in resource "vcd_nsxv_snat" "web":
 212: resource "vcd_nsxv_snat" "web" {
```
* when external netw isn't right, error isn't precise:
```
Error: unable to make structure for API call: unable to identify vNic for network 'vb_externalNe32' of type 'ext': [ENF] entity not found
```
* !!! If rule_tag specified and any changes comes from UI, when always resource will be recreated as rule_tag don't match anymore.

### PR DESCRIPTION
**Pull request by Didainius**
This PR addresses small spotted problems/inaccuracies with NSX-V related resources - `vcd_nsxv_dnat`, `vcd_nsxv_snat`, `vcd_nsxv_firewall_rule`. It closes #376 , closes #377, resolves partly #378.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,18 @@ IMPROVEMENTS:
 
 * Switch to Terraform terraform-plugin-sdk v1.0.0 as per recent [HashiCorp
   recommendation](https://www.terraform.io/docs/extend/plugin-sdk.html) - [GH-382]
+* `resource/vcd_nsxv_dnat` and `resource/vcd_nsxv_dnat` more precise error message when network is
+  not found - [GH-384]
+* `resource/vcd_nsxv_dnat` and `resource/vcd_nsxv_dnat` `rule_tag` must be int to avoid vCD internal
+  exception passthrough - [GH-384]
+* `resource/vcd_nsxv_dnat` put correct name for example in website - [GH-384]
```
Comment (lvirbalas) [inline CHANGELOG.md]: Small suggestion to increase clarity:

```suggestion
* `resource/vcd_nsxv_dnat` put correct name in doc example - [GH-384]
```

Comment (Didainius) [reply CHANGELOG.md]: Addressed.

Comment (Didainius) [reply CHANGELOG.md]: Fixed

**Comment by Didainius** [comment]
> Nit inside, also just noticed that we have supressfuncs.go and validatefuncs.go, which by our file naming convention in this project should actually be supress_funcs.go and validate_funcs.go

Fixed.

**Pull request merged**